retrofit/src/main/java/retrofit/NetworkCall.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit;
+
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.ResponseBody;
+import java.io.IOException;
+import retrofit.converter.Converter;
+
+final class NetworkCall<T> implements Call<T> {
+  private final Endpoint endpoint;
+  private final Converter converter;
+  private final OkHttpClient client;
+  private final MethodInfo methodInfo;
+  private final Object[] args;
```

Comment:
slightly annoyingly, these can be mutable, which makes executing `call()` multiple times dangerous. (Such as if this is an InputStream)